### ISSUE TITLE
[Suggestion] Lock pick Durability

### ISSUE THREAD

**Issue by Eowomir7** [Melonslise/Locks#60]
Just thought this would be a cool feature to have. i noticed someone suggested different types of lockpicks with varying break chances. This kind of expands on that idea, or rather proposes an alternative to the current % chance to break system.

**What i suggest is this: have lock picks with varying durability.** i.e. classic minecraft style wood, stone, iron, gold, diamond, netherite lock pick. but instead of having the durability be like traditional MC have it be like. wood. breaks after 1 fail, stone, breaks after 3 fails, iron breaks after 10 fails, gold breaks after 5 fails, diamond breaks after 15 fails, Netherite breaks after 20 fails. then you could possibly put unbreaking enchantment on it. these are of course just examples. not necessarily 100% balanced. just examples. 

**i feel this would be a good way to keep it balanced.** because from my own experience. the first lock i ever came across was 8 pins. it took me roughly 20 attempts to get it adding up to a total of 20 iron and leather and i was lucky on that one. i have had so many times where i go to a locked chest with a stack or more lock picks and failed to get anything. theres an exponential number of combinations for these 8 and 9 pin locks. and having the majority of my lock picks break immediately leaves me with the need for stacks and stacks of iron and leather just to open one lock. which. is far more expensive than just making a hopper.

**I believe with this type of lock pick system people will be much less encouraged to just cheat the lock.** it also gives a great progression to it, which is perfectly in line with vanilla minecraft progression, while removing the extra frustrating tediousness required to craft a ton of lock picks.

**Comment by MercurialPony** [comment]
Interesting.. I will consider this, but maybe in some future update. Currently most ways to cheat the lock have been eliminated (e.g. no more hopper shenanigans). Also more lock picks tiers have been added as you suggested

**Comment by Darkmega18** [comment]
I second this suggestion and actually wanted to make it just now also. I've had diamond lockpicks shatter on my first attempt due to terrible-terrible rng. it would be much more preferable to them to have a set amount of chances to fail and then bust, rather than occasionally bust and not bust at random. Since in that same vein I also had a bobby pick not bust twice in a row before busting.

Even if picks instead have a combination of this system.  Where you have durability, AND a chance to fail on a pick and have it be a tool rather than a loose item.

So say a wooden one has like 5 durability and doesn't break 15% of the time. So each time it "breaks" the lock resets and theres a crunch and durability goes down. 

When the durability is completely gone, kicks you out of the menu or auto-swaps you a new pick you're also holding.

Making them a tool in this way can open up to counter play against locks, for example enchanting lockpicks with unbreaking for a chance to not lose a durability when a fail occurs, or making the pick sturdy for less chance at busting when a miss is made, or something like "grace" where when it does fail it doesn't reset the tumbler but still takes durability.

something interesting to do could also be having chances to break happen more or less often based on luck attribute also.

**Comment by Eowomir7** [comment]
yeah @Darkmega18 i love your version of it. it combines the current mechanics with my version of the mechanics. really good idea. And btw this most recent patch actually adds traps and enchantments for locks. you should check out the 1.16.5 version. there's one that makes locks harder to pick. one that adds damage every time you break a lockpick, < this one could be adjusted to every time you lose a durability on the reworked lockpicks for example, or (it might be this way already idk.) whenever it kicks you out of the menu for a failed attempt. i also like the idea of Damage over time traps for locks. im gonna make a separate suggestion post for that.

**Comment by Darkmega18** [comment]
thank you. 
also 1.16.5 IS the one I'm using. I'll give it an update.

like I said, lockpicks becoming tools opens up enchanting lockpicks to counter getting dunked on by enchants so theres effort on both sides with making a strong enchanted pick to really get through mundane locks and give you half a chance of getting into enchanted ones.

I've got some other interesting ideas for you too I'll split off I think.
